Formula/singularity.rb

```diff
@@ -1,31 +1,46 @@
 class Singularity < Formula
-  desc "Application containers for Linux"
+  desc "Application container and unprivileged sandbox platform for Linux"
   homepage "https://singularity.hpcng.org"
   url "https://github.com/hpcng/singularity/releases/download/v3.8.2/singularity-3.8.2.tar.gz"
   sha256 "996611dec402b4d372b8b9456dd9ec1cb43712d08502e455f38521bd199856d3"
   license "BSD-3-Clause"
+  revision 1
 
   bottle do
     sha256 cellar: :any_skip_relocation, x86_64_linux: "565338bfa4b430b4e3de41de908e91b225e98fdcba64db857f2aedfc1c782be2"
   end
 
+  # No relocation, the localstatedir to find configs etc is compiled into the program
+  pour_bottle? only_if: :default_prefix
```

Comment:
That's the same as is defined above. `:any_skip_relocation` means it can install anywhere, it just can't be installed in a different location than it was compiled in.

Reply:
What I want is this:

"it just can't be installed in a different location than it was compiled for, else it has to be compiled"

How do I do that? I didn't see that case documented anywhere, and I found this "pour_bottle? only_if: :default_prefix"
at several places to do exactly that:

```
grep -rB2 only_if:.*:default_prefix Formula
Formula/composer.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/composer.rb-  on_macos do
Formula/composer.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
--
Formula/phpbrew.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/phpbrew.rb-  on_macos do
Formula/phpbrew.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
--
Formula/phpstan.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/phpstan.rb-  on_macos do
Formula/phpstan.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
--
Formula/pickle.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/pickle.rb-  on_macos do
Formula/pickle.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
--
Formula/psalm.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/psalm.rb-  on_macos do
Formula/psalm.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
--
Formula/easyengine.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/easyengine.rb-  on_macos do
Formula/easyengine.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
--
Formula/ocaml.rb-  # The ocaml compilers embed prefix information in weird ways that the default
Formula/ocaml.rb-  # brew detection doesn't find, and so needs to be explicitly blocked.
Formula/ocaml.rb:  pour_bottle? only_if: :default_prefix
--
Formula/wp-cli.rb-  # Keg-relocation breaks the formula when it replaces `/usr/local` with a non-default prefix
Formula/wp-cli.rb-  on_macos do
Formula/wp-cli.rb:    pour_bottle? only_if: :default_prefix if Hardware::CPU.intel?
```

So what I did should be correct.

Reply:
Notice how they're all `on_macos`, on Linux we're much better at detecting relocatability so, as the bottle block indicates, the thing you want is already the case.

Reply:
But it is still mis-executed: The bottle is downloaded and installed without recompilation for a user without sudo:

Maybe because singularity is a go program. How/Where is it implemented? 

Here is the proof:

I reconfirmed this by doing a completely fresh install in an new OS container (before, I used a system account):

I created a new container of Ubuntu 18.04 and in it, a fresh new user. As user, I ran the installation of brew from http://brew.sh

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
# BTW, this is a download of 445 MB meanwhile, it takes quite a while, the current download rate was 2 MB/s.
brew install -v singularity
# Besides this being a go
==> Summary
🍺  /home/kaindlb/.linuxbrew/Cellar/util-linux/2.37: 397 files, 19.3MB, built in 25 seconds
==> Installing singularity
==> Pouring singularity--3.8.2.x86_64_linux.bottle.tar.gz
tar --extract --no-same-owner --file /home/kaindlb/.cache/Homebrew/downloads/c723a54eb2bf05261997272b30fd3d8e5574cd61121952da4c7d1b6b516bbd3e--singularity--3.8.2.x86_64_linux.bottle.tar.gz --directory /tmp/d20210902-905-14ye14e
cp -pR /tmp/d20210902-905-14ye14e/singularity/. /home/kaindlb/.linuxbrew/Cellar/singularity
chmod -Rf +w /tmp/d20210902-905-14ye14e
==> Finishing up
ln -s ../../Cellar/singularity/3.8.2/etc/bash_completion.d/singularity singularity
mkdir -p /home/kaindlb/.linuxbrew/etc/singularity
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/capability.json capability.json
mkdir -p /home/kaindlb/.linuxbrew/etc/singularity/cgroups
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/cgroups/cgroups.toml cgroups.toml
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/ecl.toml ecl.toml
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/global-pgp-public global-pgp-public
mkdir -p /home/kaindlb/.linuxbrew/etc/singularity/network
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/network/00_bridge.conflist 00_bridge.conflist
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/network/10_ptp.conflist 10_ptp.conflist
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/network/20_ipvlan.conflist 20_ipvlan.conflist
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/network/30_macvlan.conflist 30_macvlan.conflist
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/network/40_fakeroot.conflist 40_fakeroot.conflist
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/nvliblist.conf nvliblist.conf
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/remote.yaml remote.yaml
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/rocmliblist.conf rocmliblist.conf
mkdir -p /home/kaindlb/.linuxbrew/etc/singularity/seccomp-profiles
ln -s ../../../Cellar/singularity/3.8.2/etc/singularity/seccomp-profiles/default.json default.json
ln -s ../../Cellar/singularity/3.8.2/etc/singularity/singularity.conf singularity.conf
ln -s ../Cellar/singularity/3.8.2/bin/run-singularity run-singularity
ln -s ../Cellar/singularity/3.8.2/bin/singularity singularity
==> Caveats
Bash completion has been installed to:
  /home/kaindlb/.linuxbrew/etc/bash_completion.d
==> Summary
🍺  /home/kaindlb/.linuxbrew/Cellar/singularity/3.8.2: 42 files, 157.3MB
==> Caveats
==> singularity
Bash completion has been installed to:
  /home/kaindlb/.linuxbrew/etc/bash_completion.d

```
And then, of course:
```
$ brew test singularity
==> Testing singularity
==> /home/kaindlb/.linuxbrew/Cellar/singularity/3.8.2/bin/singularity cache list
FATAL:   Couldn't not parse configuration file /home/linuxbrew/.linuxbrew/Cellar/singularity/3.8.2/etc/singularity/singularity.conf: open /home/linuxbrew/.linuxbrew/Cellar/singularity/3.8.2/etc/singularity/singularity.conf: no such file or directory
```